p2p/README.md

```diff
@@ -67,105 +66,431 @@ flowchart TD
     class PN pocket_network
 ```
 
-`Routers` is where [RainTree](https://github.com/pokt-network/pocket/files/9853354/raintree.pdf) (or the simpler basic approach) is implemented. See `raintree/router.go` for the specific implementation of RainTree, but please refer to the [specifications](https://github.com/pokt-network/pocket-network-protocol/tree/main/p2p) for more details.
+`Routers` is where [RainTree](https://github.com/pokt-network/pocket/files/9853354/raintree.pdf) is implemented.
+See [`raintree/router.go`](./raintree/router.go) for the specific implementation of RainTree, but please refer to the [specifications](https://github.com/pokt-network/pocket-network-protocol/tree/main/p2p) for more details.
 
-### Raintree Router Architecture
+## Module Architecture
 
-_DISCUSS(team): If you feel this needs a diagram, please reach out to the team for additional details._
-_TODO(olshansky, BenVan): Link to RainTree visualizations once it is complete._
+### Legends
+```mermaid
+flowchart
+subgraph Legend
+    m[[`Method`]]
+    c[Component]
+
+    m -- "unconditional usage" --> c
+    m -. "conditional usage" .-> c
+    m -. "ignored" .-x c
+end
+```
+
+```mermaid
+classDiagram
+class ConcreteType {
+  +ExportedField
+  -unexportedField
+  +ExportedMethod(argType) returnType
+  -unexportedMethod()
+}
+
+class InterfaceType {
+    <<interface>>
+    +Method(argType) (returnType1, returnType2)
+}
+
+ConcreteType --|> InterfaceType : Interface realization
+
+ConcreteType --> OtherType : Direct usage
+ConcreteType --o OtherType : Composition
+ConcreteType --* OtherType : Aggregatation
+ConcreteType ..*  "(cardinality)" OtherType : Indirect (via interface)
```

Comment:
With respect to cardinality, this example is intended to convey a 1:n, `ConcreteType`:`OtherType` relationship where "n" could be a constant or a range.

This line is mainly intended to illustrate "indirect" relation between two concrete types. The addition of cardinality was to avoid adding another relation just for that as it's getting a bit crowded already.